src/SDDPoptimize.jl

```diff
@@ -292,6 +285,13 @@ function build_model(model, param, t)
         @objective(m, Min, cost + alpha)
     end
 
+    # Add binary variable if problem is a SMIP:
+    if model.IS_SMIP
+        m.colCat[1:nx] = model.stateCat
```

Comment:
This looks very scary. Are you planning on implementing the Zou, Ahmed, Sun paper?